NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt

```diff
@@ -0,0 +1,226 @@
+        }
+
+        // Assert that back button does not exist
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertDoesNotExist()
+    }
+
+    /**
+     * Tests that the back button successfully returns to the gallery view from the item detail view
+     */
+    @Test
+    fun app_backButtonChangesPanes() {
+        composeTestRule.setContent {
+            NavigationRailAppTheme {
+                NavigationRailApp(WindowState())
+            }
+        }
+
+        // Assert "plants" title is visible
+        composeTestRule.onNodeWithText("plants").assertIsDisplayed()
+
+        // Click on first plant item
+        val firstEntryDescription =
+            composeTestRule.getString(R.string.image_description, plantList[0].name, plantList[0].id)
+        composeTestRule.onNodeWithContentDescription(firstEntryDescription).performClick()
+
+        // Assert that back button is visible
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertIsDisplayed()
+
+        // Assert that "plants" title is no longer visible
+        composeTestRule.onNodeWithText("plants").assertDoesNotExist()
+
+        // Click back button
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).performClick()
+
+        // Assert that "plants" title is visible again
+        composeTestRule.onNodeWithText("plants").assertIsDisplayed()
+    }
+
+    /**
+     * Tests that the content drawer can be swiped between the expanded and collapsed states
+     */
+    @Test
+    fun detailView_contentDrawerSwipes() {
+        composeTestRule.setContent {
+            DetailViewForFirstPlant()
+        }
+
+        // Assert drawer is collapsed on start
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer))
+            .assertDrawerStateEquals(DrawerState.Collapsed)
+
+        // Swipe content drawer up
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer)).performGesture {
+            val start = this.topCenter
+            val end = Offset(start.x, start.y - 200)
+            swipe(start, end, 200)
+        }
+
+        // Assert drawer is now expanded
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer))
+            .assertDrawerStateEquals(DrawerState.Expanded)
+
+        // Swipe content drawer back down
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer))
+            .performGesture { swipeDown() }
+
+        // Assert drawer is collapsed again
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer))
+            .assertDrawerStateEquals(DrawerState.Collapsed)
+    }
+
+    /**
+     * Tests that the info box fact contents are horizontally scrollable
+     */
+    @Test
+    fun detailView_factsAreHorizontallyScrollable() {
```

Comment:
I know it is not testing-related, but I couldn't remember why the info box has to be horizontally scrollable.

Reply:
This is mainly for the 6.7 emulator, where sometimes the second fact is cut off:
![image](https://user-images.githubusercontent.com/33138268/150427128-9b381821-ddc1-435c-8160-fbe95dd81e82.png)